vmdk_plugin/refcnt.go

```diff
@@ -203,6 +214,16 @@ func (r refCountsMap) decr(vol string) (uint, error) {
 	return rc.count, nil
 }
 
+// Return the associated mutex for the given volume
+func (r refCountsMap) getVolMtx(vol string) *sync.Mutex {
+	rc := r[vol]
+	if rc == nil {
+		rc = newRefCount()
+		r[vol] = rc
+	}
+	return rc.Mtx
+}
+
 // enumberates volumes and  builds refCountsMap, then sync with mount info
```

Comment:
Nit and optional - could add comments in the discover and sync functions that they don't use any locking anywhere when they set the count or mount/unmount volumes. They don't need to as that happens on the init path when the plugin isn't up yet.